doc/plugin/DynamicMacros.md

```diff
@@ -0,0 +1,96 @@
+# Kaleidoscope-DynamicMacros
+
+Dynamic macros are similar to [Macros][plugin:macros], but unlike them, they can
+be re-defined without compiling and flashing new firmware: one can change
+dynamic macros via [Focus][plugin:focus], using a tool like
```

Comment:
changing them via a focus plugin isn't the -only- way one can change them. Do we expose an internal API for changing them, too? So that someone could write an on-keyboard macro writing plugin?

Reply:
We do not currently expose an API to change them, the storage pointer is private, and so is the map we build on boot & change. We _could_ expose the storage pointer and size (via an accessor), which would allow a future on-keyboard macro writing plugin to update things.

Reply:
I do not think we need to add that API to merge this, but I think the doc should break out the fact that there is a Focus API for editing dynamic macros from the fact that we have these neat dynamic macros which are stored in local storage.